File: app/src/main/java/com/bytedance/tiktok/base/BaseFragment.java
package com.bytedance.tiktok.base;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.fragment.app.Fragment;

/**
 * create by libo
 * create on 2020-05-19
 * description
 */
public abstract class BaseFragment extends Fragment {
    protected View rootView;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        rootView = inflater.inflate(setLayoutId(), container, false);
        return rootView;
    }

    protected abstract int setLayoutId();

    protected abstract void init();
}

File: app/src/main/java/com/bytedance/tiktok/base/BaseFragment.kt
package com.bytedance.tiktok.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import butterknife.ButterKnife

/**
 * create by libo
 * create on 2020-05-19
 * description
 */
abstract class BaseFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        var rootView = inflater.inflate(setLayoutId(), container, false)
        ButterKnife.bind(this, rootView)
        return rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
    }

    protected abstract fun setLayoutId(): Int

    protected abstract fun init()
}